### PR TITLE
Removed No description.

### PR DESCRIPTION
**Pull request by mithlesh4257**
Fix for issue #1054
Do not show "no description found".

### CODE REVIEW

**Comment by mithlesh4257** [comment]
@nicolas-raoul  please help!

**Comment by mithlesh4257** [comment]
@maskaravivek Now is it fine? I have changed it. Now it will disable view only for description is not available.

**Comment by maskaravivek** [comment]
Yes, it should be fine now. Can you post a screenshot with this PR.

**Comment by mithlesh4257** [comment]
@maskaravivek While cloning this app to my android studio. 
Please help sir. I wrote above code as of my knowledge.
How to resolve this?

imageissues

**Comment by mithlesh4257** [comment]
@maskaravivek While cloning commons app to my android studio. I stuck here. How to fix it Sir?
After this I can proceed further?

image


issuesimage

**Comment by mithlesh4257** [comment]
@maskaravivek after cloning this app to Android Studio. The app is crashing? Can you help me solve it so that I can post Snapshot of this PR.

**Comment by mithlesh4257** [comment]
Yes @maskaravivek .
This is screeshot of it.

map
Fixed #1072
Removed No description.

**Comment by psh** [comment]
Looking at the code, I am pretty sure there's a bug: when the recycler view _reuses_ a given row there's nothing that restores the visibility of the description field; after the initial screen is visible and as you scroll down the list of nearby places, you'll never see descriptions ever again.

**Comment by mithlesh4257** [comment]
Hi @psh, Here only place description will be shown which actually has the description of it.  Not explicitly disable every description TextView.

**Comment by psh** [comment]
I pulled your branch and I tested with Longitude: **-90.1994** and Latitude: **38.6270**

There are a couple of places lacking descriptions (United States Court House and Custom House, Memorial Plaza and Episcopal Diocese of Missouri) that I expect to be hidden, and others visible.

I expected to see:

fixed

What I actually see is no descriptions at all, even for rows with good data.

broken

**Comment by misaochan** [comment]
@mithlesh4257 I glanced at your code and I suspect the bug is caused by removing `tvDesc.setText(descriptionText);` entirely outside of the `if` check. There are a few ways around this, for instance you could try setting the text for tvDesc before the `if` check?

**Comment by mithlesh4257** [comment]
Thanks alot @misaochan, @psh. It's now working fine.
updated

**Comment by neslihanturan** [comment]
There is a great experience sharing under this discussion. Thanks @psh for this explanation:)

**Comment by mithlesh4257** [comment]
Is this PR according @neslihanturan

**Comment by whym** [comment]
There seems to be a unit test to be fixed according to the change in PlaceRenderer.

The Travis-CI log contains this error:

```
org.junit.ComparisonFailure: expected:<[no description found]> but was:<[]>
        at org.junit.Assert.assertEquals(Assert.java:115)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at fr.free.nrw.commons.nearby.NearbyAdapterFactoryTest.rendererCorrectlyBoundForUnknownPlace(NearbyAdapterFactoryTest.java:99)
```

I believe you can also reproduce this by running 'gradle check'.

Since our expectation of the correct behavior has changed (from 'show "no description found"' to 'show nothing'), the test would have to be changed too.

**Comment by maskaravivek** [comment]
@mithlesh4257 Can you update the test based on @whym's suggestion

**Comment by whym** [comment]
@mithlesh4257 any update?

If we don't hear back soon, I think we can merge this. The remaining issue is a minor one and I can resolve it in a follow up commit if needed.

File: app/src/main/java/fr/free/nrw/commons/nearby/PlaceRenderer.java [hunk 1]
```diff
@@ -45,9 +45,11 @@ public void render() {
         tvName.setText(place.name);
         String descriptionText = place.getLongDescription();
         if (descriptionText.equals("?")) {
-            descriptionText = getContext().getString(R.string.no_description_found);
+            tvDesc.setText("");
+        }
+        else{
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/nearby/PlaceRenderer.java]: Sorry if this is a bit nitpicky, but our style usually utilizes a whitespace after 'else'. :)

**Comment by mithlesh4257** [comment]
Yes it's working @whym

File: app/src/main/java/fr/free/nrw/commons/nearby/PlaceRenderer.java (hunk 1 above)
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/nearby/PlaceRenderer.java]: I agree:)

**Comment by ujjwalagrawal17** [comment]
@maskaravivek I think this PR can be closed as the issue is already fixed and closed.

**Pull request closed without merging**